app/human-detector-app/services/BackendService.ts

```diff
@@ -0,0 +1,123 @@
+import axios from 'axios';
+import { executeNativeBackPress } from 'react-native-screens';
+import { z } from 'zod';
+import Group from '../classes/Group';
+import User from '../classes/User';
+import Notification from '../classes/Notification';
+import * as ServerUrl from '../config/ServerConfig';
+
+/**
+ * This file handles HTTP requests to our backend
+ */
+
+/**
+ * This method is used to get the group list that is connected to a user.  This group list
+ * will contain cameras and other information that will be stored as a group object.
+ *
+ * @param userId : the userId of the user that we need the group list for.
+ * @returns null if error occurs.
+ *          An array of Groups that are owned by the user with the userId.
+ */
+export async function getGroupListAPI(userId: string): Promise<Group[] | null> {
+  const apiLinkWithExtension: string =
+    ServerUrl.apiLink + ServerUrl.getGroupsListUrlExtension(userId);
+  let testData: any;
+  let success: boolean = false;
+  const config = {
+    headers: {
+      Accept: 'application/json',
+    },
+  };
+
+  await axios
+    .get(apiLinkWithExtension, config)
+    .then((response) => {
+      testData = response.data;
+      success = true;
+    })
+    .catch((error) => {
+      console.error(`Error code: ${error.response.status}`);
+    });
+
+  if (!success) {
+    return null;
+  }
+
+  return testData;
+}
+
+/**
+ * This method will send a notification token to the backend to be stored
+ * through our endpoints.  This is done at login.
+ *
+ * @param userIdFromLogin : userId of the user that just logged in
+ * @param expoTokenFromLogin : notification token of the user that just logged in
+ * @returns '200' string if success, else it will return the status code that backend responds with.
+ */
+export async function sendNotifyTokenAPI(
+  userIdFromLogin: string,
+  expoTokenFromLogin: string
+): Promise<string> {
```

Comment:
Axios does its own handling of errors and has a default catch if I don't give it anything.  Also, having it return the status code was just what I found the easiest to way to make the code testable (??) Axios was messing up toThrow i think with the way it returns errors as well.

`.then` catches status codes with 2xx and `.catch` catches anything else.  Default, it returns an axios error and I can't return anything in `.catch` because at that point you're in a different function.  I kind of got stuck at that so I left it as it is now.  I admit the code isn't that clean.  Let me know if you have a suggestion though because I would love to like not have it like this.

Reply:
I'll look more into this tomorrow though because like, according to the documents it should work, but it messes up with my tests which is why I went to using `.then` and `.catch`

Reply:
Hello, I was ill informed but I did some things in commit 9aaa1eedc558cbb56e064312e45ad439801c3211 please check it out